pkg/tasklet/tasklet.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasklet
+
+import (
+	"errors"
+	"os"
+	"time"
+
+	"github.com/pterm/pterm"
+)
+
+var defaultSequence = []string{"⠟", "⠯", "⠷", "⠾", "⠽", "⠻"}
+
+type StopOn struct {
+	Signal  chan os.Signal
+	Timeout time.Duration
+}
+
+func Do(startMsg string, fn func() error, endMsg string, stop StopOn) error {
```

Comment:
This is great.

Would it be worth separating out the definition of a tasklet from its execution? In the case where we want to generate then run a group of tasklets in sequence?

E.g. `tasklet.New(...tasklet.Opt)`

It might also be worth coming up with a design that allows the provider of the function to interact with the spinner as well, to allow their code to provide updates to the spinner message.

Reply:
let me know if the new design is ok..

Reply:
looks great :+1: really good point for us to iterate on as well.